File: src/main/java/com/twoguis/carfixer/model/Agenda.java
package com.twoguis.carfixer.model;

import java.sql.Date;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Agenda {
    private int id;
    private int id_horario;
    private int id_veiculo;
    private Status status;
    private Date dt_previsao;
    private Date dt_fim;
    private String observacao;
    private List<Produto> produtos;
}

File: src/app/model/agenda.ts
import { Status } from './status';

export class Agenda {
  id: number;
  id_horario: number;
  id_veiculo: number;
  status: Status;
  dt_previsao: Date|undefined;
  dt_fim: Date|undefined;
  observacao: string;
  produtos: string;

  constructor() {
    this.id = 0;
    this.id_horario = 0;
    this.id_veiculo = 0;
    this.status = Status.Pendente;
    this.dt_previsao = undefined;
    this.dt_fim = undefined;
    this.observacao = '';
    this.produtos = '';
  }
}
